src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -29,50 +29,57 @@
 using Raven.Server.Config.Categories;
 using Raven.Server.Documents;
 using Raven.Server.Documents.Indexes;
+using Raven.Server.Documents.PeriodicBackup;
 using Raven.Server.Json;
 using Raven.Server.Routing;
 using Raven.Server.ServerWide.Commands;
-using Raven.Server.ServerWide.Context;
 using Raven.Server.Smuggler.Documents.Data;
 using Raven.Server.Web.System;
 using Sparrow.Json;
 using Sparrow.Json.Parsing;
+using Sparrow.Server;
 
 namespace Raven.Server.Smuggler.Documents
 {
     public class StreamDestination : ISmugglerDestination
     {
         private readonly Stream _stream;
         private GZipStream _gzipStream;
-        private readonly DocumentsOperationContext _context;
-        private readonly DatabaseSource _source;
+        private readonly JsonOperationContext _context;
+        private readonly ISmugglerSource _source;
         private AsyncBlittableJsonTextWriter _writer;
         private DatabaseSmugglerOptionsServerSide _options;
         private Func<LazyStringValue, bool> _filterMetadataProperty;
+        public List<IDisposable> ToDispose;
 
-        public StreamDestination(Stream stream, DocumentsOperationContext context, DatabaseSource source)
+        public StreamDestination(Stream stream, JsonOperationContext context, ISmugglerSource source)
         {
             _stream = stream;
             _context = context;
             _source = source;
+            ToDispose ??= new List<IDisposable>();
         }
 
-        public IAsyncDisposable InitializeAsync(DatabaseSmugglerOptionsServerSide options, SmugglerResult result, long buildVersion)
+        public IAsyncDisposable InitializeAsync(DatabaseSmugglerOptionsServerSide options, SmugglerResult result, long buildVersion, bool sharded = false)
```

Comment:
remove the default value from here, this is internal API and not used in a lot of places, will help you to find all the places where you need to set it correctly